### PR TITLE
New operative reinforcement option: Intelligence Overwatch Agent

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

Introducing a new Nuclear Operative reinforcement option: The Overwatch Intelligence Agent.

Equipped with multi-hudglasses, they have an advanced camera console, station alerts, and bodycams of every operative! If something can be known, they will know about it.

They can also remotely pilot your ship. Finally, everyone can ride in the Steel Rain without getting stuck on the station! 

This role spawns in the formerly unused outpost just north of the nukie base. With a few shelves of supplies and some tools in the back room, they can set up their workplace however they like. This also gives them something to work on while they wait for the operatives to gear up.

image

As you can see, it's rather cramped and the lights are quite dim in the backroom. Set it up however you like, this is how I did mine:

image

Total price? 12 TC per agent. It might get a bit cramped, but you could buy a second to make sure the first guy doesn't get lonely!

This turned into a 30-commit ugly because the bodycams were originally meant to be accomplished via a refactoring of the spyglass kit. Big mistake that made me shelve the project -- until Melbert's simple bodycam component conveniently did exactly what I needed in a much simpler way.
## Why It's Good For The Game

Having a "guy in the chair" for your kickass murder operator squad enables more brainy strategizing, and is thematically sound. Also, nukies have the opportunity to bring in another player to participate in the fun!
## Changelog
:cl: Rhials
add: Nuclear Operatives now purchase an Intelligence Agent, who can watch cameras and bodycams, move the shuttle, and provide radio support. Only 12 Telecrystals!
/:cl:

### CODE REVIEW

File: code/__DEFINES/antagonists.dm [hunk 1]
```diff
@@ -397,3 +397,6 @@ GLOBAL_LIST_INIT(human_invader_antagonists, list(
 #define SPY_DIFFICULTY_MEDIUM "Medium"
 /// Very difficult to accomplish, almost guaranteed to require crew conflict
 #define SPY_DIFFICULTY_HARD "Hard"
+
+/// Camera network nuclear operative bodycams are on
+#define OPERATIVE_CAMERA_NET "operative"
```
Comment (JohnFulpWillard) [inline code/__DEFINES/antagonists.dm:402]: Put this in the file with all the other camera nets please

File: code/modules/antagonists/nukeop/datums/operative_support.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		c_tag = "[new_teammate.real_name]", \
+		network = OPERATIVE_CAMERA_NET, \
+		emp_proof = FALSE, \
+	)
+	to_chat(new_teammate, span_notice("You have been equipped with a bodycam, viewable by your Overwatch Intelligence Agent. Make sure to show them a good performance!"))
+
+/datum/objective/overwatch
+
+/datum/objective/overwatch/New()
+	explanation_text = "Provide intelligence support and overwatch to your operative team!"
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/nukeop/datums/operative_support.dm:51]: Do we need to set this on New for any reason in particular?

File: code/modules/antagonists/nukeop/datums/operative_support.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	owner.current.playsound_local(get_turf(owner.current), 'sound/machines/printer.ogg', 100, 0, use_reverb = FALSE)
+	to_chat(owner, span_big("You are a [name]! You've been temporarily assigned to provide camera overwatch and manage communications for a nuclear operative team!"))
+	to_chat(owner, span_red("Use your tools to set up your equipment however you like, but do NOT attempt to leave your outpost."))
+	owner.announce_objectives()
+
+/datum/antagonist/nukeop/support/on_gain()
+	..()
+	for(var/datum/mind/teammate_mind in nuke_team.members)
+		var/mob/living/our_teammate = teammate_mind.current
+		our_teammate.AddComponent( \
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/nukeop/datums/operative_support.dm:17]: We're not checking if they've already got one before giving them another, yeah it's a component but i dunno.

File: code/__DEFINES/antagonists.dm (hunk 1 above)
Comment (Rhials) [reply code/__DEFINES/antagonists.dm:402]: Do we actually have a file with all the camera nets in the same place? __DEFINES/camnets.dm doesn't actually have any cam nets and the recent bitrunning camnet was in the bitrunning defines. Should I just move those two to camnets.dm?

File: code/modules/antagonists/nukeop/datums/operative_support.dm (hunk 2 above)
Comment (Rhials) [reply code/modules/antagonists/nukeop/datums/operative_support.dm:51]: That's just an artifact of a plan that never happened. No need.

File: code/modules/antagonists/nukeop/datums/operative_support.dm (hunk 3 above)
Comment (MrMelbert) [reply code/modules/antagonists/nukeop/datums/operative_support.dm:17]: that component is `COMPONENT_DUPE_HIGHLANDER` anyways (though i guess it should be `COMPONENT_DUPE_SELECTIVE` to only allow dupes on different networks, i didn't think of that)

File: code/__DEFINES/antagonists.dm (hunk 1 above)
Comment (JohnFulpWillard) [reply code/__DEFINES/antagonists.dm:402]: I added it in the PR that removed camera assembly structures

Comment (Rhials) [reply code/__DEFINES/antagonists.dm:402]: Ahh, couldn't find it because my branch wasn't up to date.

**Comment by moocowswag** [comment]
I fear for anyone that gets this role that doesnt know how to make machines/consoles somehow.

**Comment by Admiral-Gigaton** [comment]
bolted door, stopping them from getting to the machine parts??

**Comment by SkeletalElite** [comment]
Give this role codespeak in case the operatives decide to buy a codespeak book to avoid ease dropping

**Comment by Rhials** [comment]
> bolted door, stopping them from getting to the machine parts??

the door is very much not bolted you can walk through it by clicking or walking into it

**Comment by Deadgebert** [comment]
Can they get a loud mode headset?

**Comment by Rhials** [comment]
> Can they get a loud mode headset? It's hard to parse the comms mid combat, it'll be easier to listen to them if the text is bigger

They already get one in their outfit!

**Comment by junkgle01** [comment]
this is great, i've always thought nukies needed more guiding because when everyone gets disorganized it all goes to shit. i love it

**Comment by Singul0** [comment]
simple food amenities would be neat

**Comment by Rhials** [comment]
> simple food amenities would be neat

One shelf is dedicated to providing the nourishment necessary for their job (donuts and syndie-cakes).

**Comment by optimumtact** [comment]
wow a harmyeller

**Comment by Rhials** [comment]
> Very neat. Although it is a bit pricey for just intelligence. 8 would make it a bit more tempting I think. Also would be better to start them off pre-built, so that they can use the time nukies are setting up to spy on the station and get some intel on vulnerable areas, strategic landing points etc.

I went with 12 TC so it's not a complete write-off for smaller nuke squads or non-war ops. War ops won't bat an eye at 12 TC, and they're going to be the ones who need it the most.

I have no intent to make the area is set up already. It's part of the job, just like moving surgical tables to the op room in medical or moving around your botany trays at the start of a round. A spaceman with quick fingers could realistically get the whole area set up in under two minutes if it's that urgent.

**Pull request merged**